src/routes/courierPage.tsx

```diff
@@ -1,314 +1,314 @@
-
-export const Route = createFileRoute("/courierPage")({
-  component: RouteComponent,
-});
-
-type UserOption = {
-  value: string;
-  label: string;
-};
-
-type CompartmentOption = {
-  id: string;
-  compartment: string | number; // Updated to handle both string and number
-};
-
-function RouteComponent() {
-  const [users, setUsers] = useState<UserOption[]>([]);
-  const [selectedUser, setSelectedUser] =
-    useState<SingleValue<UserOption>>(null);
-  const [compartmentSizes] = useState([
-    { value: "Extra Large", label: "Extra Large" },
-    { value: "Large", label: "Large" },
-    { value: "Medium", label: "Medium" },
-    { value: "Small", label: "Small" },
-  ]);
-  const [selectedSize, setSelectedSize] =
-    useState<SingleValue<{ value: string; label: string }>>(null);
-  const [availableCompartments, setAvailableCompartments] = useState<
-    CompartmentOption[]
-  >([]);
-  const [selectedCompartment, setSelectedCompartment] = useState<string>("");
-  const [selectedCourier, setSelectedCourier] = useState<string>("Amazon");
-
-  const db = getFirestore();
-
-  // Fetch users from Firestore
-  useEffect(() => {
-    const fetchUsers = async () => {
-      const usersCollection = collection(db, "users");
-      const userDocs = await getDocs(usersCollection);
-      const userList = userDocs.docs.map((doc) => ({
-        value: doc.id,
-        label: doc.data().username,
-      }));
-      setUsers(userList);
-    };
-
-    fetchUsers();
-  }, [db]);
-
-  // Fetch compartments when a size is selected
-  useEffect(() => {
-    if (!selectedSize) return;
-
-    const fetchCompartments = async () => {
-      try {
-        const vaultsCollection = collection(db, "vault");
-        const compartmentsQuery = query(
-          vaultsCollection,
-          where("packsize", "==", selectedSize.value),
-          where("status", "==", "available")
-        );
-        const compartmentDocs = await getDocs(compartmentsQuery);
-
-        const compartments = compartmentDocs.docs.map((doc) => ({
-          id: doc.id,
-          compartment: doc.data().compartment,
-        }));
-
-        setAvailableCompartments(compartments);
-      } catch (error) {
-        Swal.fire({
-          icon: "error",
-          title: "Error",
-          text: "Failed to fetch compartments. Please try again later.",
-        });
-      }
-    };
-
-    fetchCompartments();
-  }, [selectedSize, db]);
-
-  const handleSubmit = async () => {
-    console.log("Selected User:", selectedUser);
-    console.log("Selected Size:", selectedSize);
-    console.log("Selected Compartment:", selectedCompartment);
-    console.log("Selected Courier:", selectedCourier);
-
-    if (
-      !selectedUser ||
-      !selectedSize ||
-      !selectedCompartment ||
-      !selectedCourier
-    ) {
-      Swal.fire({
-        icon: "error",
-        title: "Missing Fields",
-        text: "Please fill in all the fields before submitting!",
-      });
-      return;
-    }
```

Comment:
Improve validation here, take a look into zod and the examples I did in class https://zod.dev/